packages/devtools_app/lib/src/screens/inspector_v2/inspector_tree_controller.dart

```diff
@@ -187,78 +183,99 @@ class InspectorTreeController extends DisposableController
   set selection(InspectorTreeNode? node) {
     if (node == _selection) return;
 
-    setState(() {
-      _selection?.selected = false;
-      _selection = node;
-      _selection?.selected = true;
-      final configLocal = config;
-      if (configLocal.onSelectionChange != null) {
-        configLocal.onSelectionChange!();
-      }
-    });
+    _selection?.selected = false;
+    _selection = node;
+    _selection?.selected = true;
+    final configLocal = config;
+    if (configLocal.onSelectionChange != null) {
+      configLocal.onSelectionChange!();
+    }
+    _updateRows();
   }
 
   InspectorTreeNode? get hover => _hover;
   InspectorTreeNode? _hover;
 
   double? lastContentWidth;
 
-  final cachedRows = <InspectorTreeRow?>[];
   InspectorTreeRow? _cachedSelectedRow;
 
   /// All cached rows of the tree.
   ///
-  /// Similar to [cachedRows] but:
+  /// Similar to [rowsInTree] but:
   /// * contains every row in the tree (including collapsed rows)
   /// * items don't change when nodes are expanded or collapsed
   /// * items are populated only when root is changed
   final _searchableCachedRows = <InspectorTreeRow?>[];
 
-  void setSearchTarget(SearchTargetType searchTarget) {
-    _searchTarget = searchTarget;
-    refreshSearchMatches();
+  /// All the rows that should be displayed in the tree.
+  ///
+  /// The rows can be updated with a call to [_updateRows].
+  ValueListenable<List<InspectorTreeRow?>> get rowsInTree => _rowsInTree;
+  final _rowsInTree = ListValueNotifier<InspectorTreeRow?>([]);
```

Comment:
since we are only using the `replaceAll` method, let's just use a ValueNotifier instead of a ListValueNotifier, since we aren't really getting the added value from ListValueNotifier in this case if we are always replacing the list.

Reply:
It looks like the one benefit `ListValueNotifer` (instead of calling `clear` then `addAll`) is listeners will only be notified once instead of twice.

Reply:
Chatted offline, switched to using `ValueNotifier`